src/askvm/lib/run.ts

```diff
@@ -34,9 +39,9 @@ export async function run(
     return typed(code);
   }
 
-  ops += 1;
-  if (ops > 1000) {
-    // throw new Error('Over ops limit!');
+  options.stats!.steps += 1;
+  if (options.stats!.steps >= options.limits!.steps) {
+    throw new Error('Over steps limit!');
```

Comment:
@mhagmajer , could you list all operations that are considered a step?
It's important to make sure we don't interrupt valid computations too early.

Reply:
Every resource reference is a step. Also computing resources usually requires making more steps, for example:
- https://github.com/xFAANG/askql/blob/master/src/askvm/resources/core/if.ts
- https://github.com/xFAANG/askql/blob/master/src/askvm/resources/core/while.ts
- https://github.com/xFAANG/askql/blob/master/src/askvm/resources/core/query.ts

Reply:
OK, sounds good.
And how does fetching results in `query` count? 
E.g. if I wanted to get an aggregated query to sum people income in my corporation and run:
```
const people = query {
   person {
     income
   }
}
people:sum()
```
for a list of 10 001 people, would it fail?
How could I get the result I need anyway?

Reply:
In this PR I only attempted to fix the issue with infinite loops. Configuring the limit will come next because it requires some changes to `extendOptions`

Reply:
Configuring the limit is another topic. 
What I wanted to ask in the comment above is how `query` operations are counted?

Reply:
I just checked that `query` uses `run` twice for each node - one for name and one for value. Maybe we can take a closer look at it and its performance on another issue once this is closed?

Reply:
Sounds good.  Filing as https://github.com/xFAANG/askql/issues/303